### PR TITLE
http_server add stall_handler, it just stall the connection.

### PR DESCRIPTION
**Pull request by tnextday**
http_server add stall_handler, it just stall the connection.

### CODE REVIEW

**Comment by tnextday** [comment]
The new version of clang will have compile errors.
```
clang: error: optimization flag '-finline-functions' is not supported
```

File: src/http_server.cpp [hunk 1]
```diff
@@ -301,20 +305,28 @@ namespace sim
...
 
+		m_recv_buffer.erase(m_recv_buffer.begin(), m_recv_buffer.begin() + req_len);
+		m_bytes_used -= req_len;
+
 		auto it = m_handlers.find(req.path);
 		if (it == m_handlers.end())
 		{
-			// no handler found, 404
-			m_send_buffer = send_response(404, "Not Found");
+			if (std::find(m_stall_handlers.begin(),
```
Comment (arvidn) [inline src/http_server.cpp]: in which case this becomes: ``if (m_stall_handlers.count(req.path)) return;``
no need to indent the the 404 response or wrap it in an else-block

**Comment by arvidn** [comment]
looks good, apart from one comment

**Comment by arvidn** [comment]
I'll try to fix the command line option error separately. boost-build probably needs updating

**Comment by arvidn** [comment]
If you rebase this on top of master the build error should be fixed

**Pull request merged**